decode_test.go

```diff
@@ -1183,39 +1137,23 @@ func TestUnmarshalerPointerField(t *testing.T) {
 	for _, item := range unmarshalerTests {
 		obj := &unmarshalerPointer{}
 		err := yaml.Unmarshal([]byte(item.data), obj)
-		if err != nil {
-			t.Fatalf("Unmarshal() returned error: %v", err)
-		}
+		assert.NoError(t, err)
 		if item.value == nil {
-			if obj.Field != nil {
-				t.Fatalf("Unmarshal() returned %v, want nil", obj.Field)
-			}
+			assert.IsNil(t, obj.Field)
 		} else {
-			if obj.Field == nil {
-				t.Fatalf("Unmarshal() returned nil, want non-nil. Value: %#v", item.value)
-			}
-			if !reflect.DeepEqual(obj.Field.value, item.value) {
-				t.Fatalf("Unmarshal() returned\n%#v\nbut expected\n%#v", obj.Field.value, item.value)
-			}
+			assert.NotNil(t, obj.Field, "Pointer not initialized (%#v)", item.value)
```

Comment:
You didn't work on having NotNilf

https://github.com/yaml/go-yaml/pull/66#discussion_r2279739002

Reply:
It's done